crow/recipes/binned_exact.py

```diff
@@ -60,45 +60,19 @@ def _setup_with_completeness(self):
             self._completeness_distribution = self.completeness.distribution
 
     def _setup_with_purity(self):
-        """Makes mass distribution use additional integral with completeness"""
-        if self.purity is None:
+        """Makes mass distribution use additional integral with purity"""
+        if self.purity is None: 
             self._mass_distribution_distribution = self.mass_distribution.distribution
         else:
             self._mass_distribution_distribution = self._impure_mass_distribution
 
-    def _impure_mass_distribution(self, log_mass, z, log_mass_proxy_limits):
+    def _impure_mass_distribution(self, log_mass, z, log_mass_proxy):
```

Comment:
This here is the big problem for all the functions right? Because the impure does not take only a tuple but an NDArray of log_mass_proxy